lib/application.ts

```diff
@@ -1573,6 +1574,12 @@ export class SNApplication {
 
   private createHttpManager() {
     this.httpService = new SNHttpService();
+    this.serviceObservers.push(this.httpService.addEventObserver((tag) => {
+      if (tag === ErrorTag.RevokedSession) {
+        /** Sign out without warning. */
```

Comment:
Hmm, thinking about "without warning"..

Reply:
I think in the end if you're revoking a session you could be defending yourself against an attacker, and giving them a warning before destroying the data would give them a window to back it up, thus defeating the purpose of the destruction mechanism.

Then maybe revoking a session needs to be an authenticated action (as in, prompts for password/passcode/biometrics).